x-pack/legacy/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -481,39 +480,43 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
-                              'No fields match the current filters. Try changing your filters or time range.',
-                          })
-                        : i18n.translate('xpack.lens.indexPatterns.noFieldsLabel', {
-                            defaultMessage: 'No fields exist in this index pattern.',
-                          })
+                    localState.typeFilter.length || localState.nameFilter.length
+                      ? i18n.translate('xpack.lens.indexPatterns.noFilteredFieldsLabel', {
+                          defaultMessage: 'No fields match the current filters.',
+                        })
+                      : showEmptyFields
+                      ? i18n.translate('xpack.lens.indexPatterns.noFieldsLabel', {
+                          defaultMessage: 'No fields exist in this index pattern.',
+                        })
                       : i18n.translate('xpack.lens.indexPatterns.emptyFieldsWithDataLabel', {
-                          defaultMessage:
-                            'No fields have data with the current filters and time range. Try changing your filters or time range.',
+                          defaultMessage: 'Looks like you don’t have any data.',
                         })
                   }
                 >
                   {(!showEmptyFields ||
                     localState.typeFilter.length ||
                     localState.nameFilter.length) && (
-                    <EuiButtonEmpty
-                      size="xs"
-                      color="primary"
-                      flush="left"
-                      data-test-subj="lnsDataPanelShowAllFields"
-                      onClick={() => {
-                        trackUiEvent('indexpattern_show_all_fields_clicked');
-                        clearLocalState();
-                        onToggleEmptyFields(true);
-                      }}
-                    >
-                      {i18n.translate('xpack.lens.indexPatterns.showAllFields.buttonText', {
-                        defaultMessage: 'Show all fields',
-                      })}
-                    </EuiButtonEmpty>
+                    <>
+                      <strong>
+                        {i18n.translate('xpack.lens.indexPatterns.noFields.tryText', {
+                          defaultMessage: 'Try:',
+                        })}
+                      </strong>
+                      <ul>
+                        <li>
+                          {i18n.translate('xpack.lens.indexPatterns.noFields.extendTimeBullet', {
+                            defaultMessage: 'Extending the time range',
+                          })}
+                        </li>
+                        <li>
+                          {i18n.translate('xpack.lens.indexPatterns.noFields.fieldFilterBullet', {
+                            defaultMessage:
+                              'Using the Filter by type option {arrow} to show fields without data',
```

Comment:
`Filter by type` is a translated string as well, with the id `xpack.lens.indexPatterns.toggleFiltersPopover`. Since it's a button label, it would be worth reusing the same translation by passing it in as a `value`, for example:

```
i18n.translate('...', {
  defaultMessage: 'Using the {filterByType} button',
  values: {
    filterByType: i18n.translate('xpack.lens.indexPatterns.toggleFiltersPopover', defaultMessage: 'Filter by type'),
  }
}
```

Reply:
I ended up just creating a stored var for this translation and re-using it in the proper places.